components/layout_2020/table/layout.rs

```diff
@@ -1661,6 +1687,72 @@ impl<'a> TableLayout<'a> {
             )));
         }
     }
+
+    fn compute_border_collapse(&mut self, writing_mode: WritingMode) {
+        if self.table.style.get_inherited_table().border_collapse != BorderCollapse::Collapse {
+            self.collapsed_borders = None;
+            return;
+        }
+
+        let mut collapsed_borders = CollapsedBorders {
+            block: vec![Au::zero(); self.table.size.height + 1],
+            inline: vec![Au::zero(); self.table.size.width + 1],
+        };
+
+        for row_index in 0..self.table.size.height {
+            for column_index in 0..self.table.size.width {
+                let cell = match self.table.slots[row_index][column_index] {
+                    TableSlot::Cell(ref cell) => cell,
+                    _ => continue,
+                };
+
+                let border = cell.style.border_width(writing_mode);
+                collapsed_borders.block[row_index].max_assign(border.block_start.into());
+                collapsed_borders.block[row_index + cell.rowspan]
+                    .max_assign(border.block_end.into());
+                collapsed_borders.inline[column_index].max_assign(border.inline_start.into());
+                collapsed_borders.inline[column_index + cell.colspan]
+                    .max_assign(border.inline_end.into());
+            }
+        }
+
+        self.collapsed_borders = Some(collapsed_borders);
+    }
+
+    fn get_collapsed_borders_for_cell(
+        &self,
+        cell: &TableSlotCell,
+        coordinates: TableSlotCoordinates,
+    ) -> Option<LogicalSides<Length>> {
+        let Some(ref collapsed_borders) = self.collapsed_borders else {
+            return None;
+        };
+
+        let end_x = coordinates.x + cell.colspan;
+        let end_y = coordinates.y + cell.rowspan;
+        let mut result: LogicalSides<Length> = LogicalSides {
+            inline_start: collapsed_borders.inline[coordinates.x],
+            inline_end: collapsed_borders.inline[end_x],
+            block_start: collapsed_borders.block[coordinates.y + 1],
```

Comment:
The `+1` seems wrong.

Reply:
Yep. I've corrected this in the latest version.